blockchain_connector/Dockerfile

```diff
@@ -73,30 +63,36 @@ RUN OPENSSL_VER=1.1.1d \
  && make test \
  && make install -j$THREADS
 
+
 # Build Avalon blockchain connector intermediate docker image
 FROM base_image as build_image
 # Add necessary packages to build
 RUN apt-get update \
  && apt-get install -y -q \
+    pkg-config \
+    protobuf-compiler \
+    libprotobuf-dev \
+    software-properties-common \
+    dh-autoreconf \
+    ocaml \
+    xxd \
+    ocamlbuild \
     cmake \
     make \
     git \
-    wget \
-    tar \
-    unzip \
     swig \
  && apt-get -y -q upgrade \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
 
-ENV TCF_HOME=/project/avalon
 
-COPY VERSION /project/avalon/
-COPY ./bin /project/avalon/bin
-COPY ./common/cpp /project/avalon/common/cpp
-COPY ./common/python /project/avalon/common/python
-COPY ./sdk /project/avalon/sdk
-COPY ./blockchain_connector /project/avalon/blockchain_connector
+# Installing web3 for ethereum
+RUN pip3 install --upgrade setuptools json-rpc web3 py-solc-x nose2 eth-abi
+
+# Install specific version of solc to support versions <0.6.0
+RUN python3 -m solcx.install v0.5.15
+
+RUN pip3 install nest_asyncio
```

Comment:
pip packages like json-rpc web3 py-solc-x nose2 eth-abi is also needed at run time and should ideally be installed in base image. I guess things are working now because of below line in final image
COPY --from=build_image /usr/local/ /usr/local. This approach also is not optimal. We should be copy only required packages in final image instead of copying entire /usr/local

Reply:
To be handled in a seperate PR. Issue raised https://github.com/hyperledger/avalon/issues/307 to track this.